File: R/8kyu/8kyu-sum-mixed-array.r
# Sum Mixed Array
# Langugage: R
# Version: 3.4.1
# Difficulty: 8 kyu

sum_mix <- function(vec){
  sum(as.numeric(vec))
}


# Tests
test_that('Basic tests', {
  expect_equal(sum_mix(c(9, 3, '7', '3')), 22)
  expect_equal(sum_mix(c('5', '0', 9, 3, 2, 1, '9', 6, 7)), 42)
  expect_equal(sum_mix(c('3', 6, 6, 0, '5', 8, 5, '6', 2,'0')), 41)
  expect_equal(sum_mix(c('1', '5', '8', 8, 9, 9, 2, '3')), 45)
  expect_equal(sum_mix(c(8, 0, 0, 8, 5, 7, 2, 3, 7, 8, 6, 7)), 61)
})

File: Python/8kyu/8kyu-sum-mixed-array.py
# Sum Mixed Array
# Langugage: Python
# Version: 3.10
# Difficulty: 8 kyu

# Solution 1
def sum_mix(arr):
    return sum([int(n) for n in arr])

# Refactored Solution 1
def sum_mix(arr):
    return sum(int(n) for n in arr)

# Solution 2
def sum_mix(arr):
    return sum(map(int, arr))


# Tests
import codewars_test as test
from solution import sum_mix

@test.describe("Fixed Tests")
def fixed_tests():
    @test.it('Basic Test Cases')
    def basic_test_cases():
        test.assert_equals(sum_mix([9, 3, '7', '3']), 22)
        test.assert_equals(sum_mix(['5', '0', 9, 3, 2, 1, '9', 6, 7]), 42)
        test.assert_equals(sum_mix(['3', 6, 6, 0, '5', 8, 5, '6', 2,'0']), 41)
        test.assert_equals(sum_mix(['1', '5', '8', 8, 9, 9, 2, '3']), 45)
        test.assert_equals(sum_mix([8, 0, 0, 8, 5, 7, 2, 3, 7, 8, 6, 7]), 61)